packages/hardhat-core/src/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -612,18 +622,19 @@ export class EthModule {
           "blockHash is mutually exclusive with fromBlock/toBlock"
         );
       }
+
       const block = await this._node.getBlockByHash(filter.blockHash);
       if (block === undefined) {
         throw new InvalidArgumentsError("blockHash cannot be found");
       }
 
-      filter.fromBlock = new BN(block.header.number);
-      filter.toBlock = new BN(block.header.number);
+      filter.fromBlock = block.header.number;
+      filter.toBlock = block.header.number;
     }
 
     const [fromBlock, toBlock] = await Promise.all([
-      this._extractBlock(filter.fromBlock),
-      this._extractBlock(filter.toBlock),
+      this._normalizeOldBlockTagForFilterRequest(filter.fromBlock),
```

Comment:
These blocktags aren't resolved, but rather normalized, because the filter-related code is a bit different.